publish/publish.go

```diff
@@ -0,0 +1,75 @@
+package publish
+
+import (
+	"strconv"
+
+	"github.com/grafana/metrictank/metrics_client"
+	schema "github.com/grafana/metrictank/schema"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	ingestedMetrics = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "gateway",
+		Name:      "samples_ingested_total",
+		Help:      "Number of samples ingested",
+	}, []string{"org"})
+)
+
+type Publisher interface {
+	Publish(metrics []*schema.MetricData) error
+	Type() string
+}
+
+var (
+	publisher Publisher
+
+	// Persister allows pushing metrics to the Persistor Service
+	Persistor *metrics_client.Client
```

Comment:
It doesn't look like this `Persistor` is actually used (either in this PR or anywhere in the tsdb-gw project), which I believe means we can drop it as well as the `metrics_client` package which was pulled in specifically for this.

Reply:
hmm i have a feeling that this was added because historically we used to use tsdb-gw also to ingest data into cortex, but now we don't do that anymore afaik

Reply:
so `metrics_client` is only necessary for the `persistor`? in that case i think we can remove all of it probably

Reply:
3af36e930d997bc3facbc9d115d4ba5401c00ed5